File: Lab1_2/C++/Class.cpp
#include "Class.h"

// Конструктор без параметрів
StringClass::StringClass() {
    text = "";
}

// Конструктор з параметром
StringClass::StringClass(const std::string& s) {
    text = s;
}

// Конструктор копіювання
StringClass::StringClass(const StringClass& other) {
    text = other.text;
}

// Деструктор
StringClass::~StringClass() {
}

// Довжина рядка
int StringClass::Length() const {
    return text.length();
}

// Зсув
void StringClass::ShiftRight() {
    if (text.length() > 1) {
        char last = text.back();
        text.pop_back();
        text = last + text;
    }
}

// Отримати рядок
std::string StringClass::GetText() const {
    return text;
}

File: Lab1_2/C#/Class.cs
using System;

namespace StringLibrary
{
    public class StringClass
    {
        private string text; // прихований рядок

        // Конструктор без параметрів
        public StringClass()
        {
            text = "";
        }

        // Конструктор з параметром
        public StringClass(string s)
        {
            text = s;
        }

        // Конструктор копіювання
        public StringClass(StringClass other)
        {
            text = other.text;
        }

        // Деструктор
        ~StringClass()
        {
            // Garbage Collector
        }

        // Довжина
        public int Length()
        {
            return text.Length;
        }

        // Зсув
        public void ShiftRight()
        {
            if (text.Length > 1)
            {
                char last = text[text.Length - 1];
                text = last + text.Substring(0, text.Length - 1);
            }
        }

        // Отримати рядок
        public string GetText()
        {
            return text;
        }
    }
}